From ff52bc1578af95ea651a0f4ad160e0e2ec6a6ae4 Mon Sep 17 00:00:00 2001
From: Flora Cui <Flora.Cui@amd.com>
Date: Mon, 19 Sep 2016 17:26:27 +0800
Subject: [PATCH 1579/2094] drm/amdgpu: unbind gart when release direct gma
 imported bo

Change-Id: Ie8db112202ad2b11a5ec1294dc9b4163eaa0a542
Signed-off-by: Flora Cui <Flora.Cui@amd.com>
Reviewed-by: Hawking Zhang <Hawking.Zhang@amd.com>

Conflicts:
	drivers/gpu/drm/amd/amdgpu/amdgpu_object.c
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_object.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_object.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_object.c
index 92ed03a..4347177 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_object.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_object.c
@@ -102,9 +102,15 @@ static void amdgpu_ttm_bo_destroy(struct ttm_buffer_object *tbo)
 {
 	struct amdgpu_device *adev = amdgpu_ttm_adev(tbo->bdev);
 	struct amdgpu_bo *bo;
+	u64 offset;
 
 	bo = container_of(tbo, struct amdgpu_bo, tbo);
 
+	if (bo->tbo.mem.mem_type == AMDGPU_PL_DGMA_IMPORT) {
+		offset = amdgpu_bo_gpu_offset(bo);
+		offset -= adev->mman.bdev.man[TTM_PL_TT].gpu_offset;
+		amdgpu_gart_unbind(adev, offset, bo->tbo.num_pages);
+	}
 	amdgpu_update_memory_usage(adev, &bo->tbo.mem, NULL);
 
 	drm_gem_object_release(&bo->gem_base);
-- 
2.7.4

